src/routes/privs/__init__.py

```diff
@@ -11,18 +11,24 @@
 
 @privs_blueprint.route('/privs/has-data-admin')
 def privs_has_data_admin_privs():
-    groups_token: str = get_groups_token()
-    auth_helper_instance: AuthHelper = AuthHelper.instance()
-
-    data_admin_privs: List[dict] = auth_helper_instance.has_data_admin_privs(groups_token)
-    if isinstance(data_admin_privs, Response):
-        return data_admin_privs
-
+    """
+    The endpoint will return a status_code of 401 (as per the ingest-api-spec.yaml file)
+    if the "User's token is not valid" (missing, or not logged in).
+    For a valid logged in token it will return the json {'has_data_admin_privs': true/false}
+    with a status_code of 200.
+    """
     headers: dict = {
         "Content-Type": "application/json"
     }
-    return make_response(jsonify({"has_data_admin_privs": data_admin_privs}), 200, headers)
+    start_of_token: int = len('BEARER ')
 
+    authorization: str = request.headers.get('authorization')
+    if authorization is None or len(authorization) < start_of_token:
+        return Response("Non-active login", 401)
 
-def get_groups_token() -> str:
-    return request.headers.get('authorization')[7:]
+    groups_token: str = authorization[start_of_token:]
+    auth_helper_instance: AuthHelper = AuthHelper.instance()
+    data_admin_privs: List[dict] = auth_helper_instance.has_data_admin_privs(groups_token)
+    if isinstance(data_admin_privs, Response):
+        return data_admin_privs
+    return make_response(jsonify({"has_data_admin_privs": data_admin_privs}), 200, headers)
```

Comment:
Using `jsonify` should be sufficient, and it'll remove the need to specify the `headers`

```
    headers: dict = {
        "Content-Type": "application/json"
    }
```